src/main/java/it/gov/pagopa/admissibility/service/onboarding/AuthoritiesDataRetrieverServiceImpl.java

```diff
@@ -47,45 +71,80 @@ public Mono<OnboardingDTO> retrieve(OnboardingDTO onboardingRequest, InitiativeC
-                            ? Residence.builder()
-                                .city("Milano")
-                                .cityCouncil("Milano")
-                                .province("Milano")
-                                .region("Lombardia")
-                                .postalCode("20124")
-                                .nation("Italia")
-                                .build()
-                            : Residence.builder()
-                                .city("Roma")
-                                .cityCouncil("Roma")
-                                .province("Roma")
-                                .region("Lazio")
-                                .postalCode("00187")
-                                .nation("Italia")
-                                .build()
-
-            );
-        }
-        if(onboardingRequest.getBirthDate()==null && is2retrieve(initiativeConfig, OnboardingConstants.CRITERIA_CODE_BIRTHDATE)) {
-            int age = userIdBasedIntegerGenerator(onboardingRequest).nextInt(18, 99);
-            onboardingRequest.setBirthDate(BirthDate.builder()
-                    .age(age)
-                    .year((LocalDate.now().getYear()-age)+"")
-                    .build());
+        PdndServicesInvocation pdndServicesInvocation = new PdndServicesInvocation(
+        onboardingRequest.getIsee() == null && is2retrieve(initiativeConfig, OnboardingConstants.CRITERIA_CODE_ISEE),
+        onboardingRequest.getResidence() == null && is2retrieve(initiativeConfig, OnboardingConstants.CRITERIA_CODE_RESIDENCE),
+        onboardingRequest.getBirthDate() == null && is2retrieve(initiativeConfig, OnboardingConstants.CRITERIA_CODE_BIRTHDATE)
+        );
+
+        if (pdndServicesInvocation.requirePdndInvocation()) {
+            return  createTokenService.getToken(initiativeConfig.getPdndToken())
+                    .zipWith(userFiscalCodeService.getUserFiscalCode(onboardingRequest.getUserId()))
+                    .flatMap(t -> invokePdndServices(t.getT1(), onboardingRequest, t.getT2(), pdndServicesInvocation, message));
         }
+
         return Mono.just(onboardingRequest);
     }
 
-    private static Random userIdBasedIntegerGenerator(OnboardingDTO onboardingRequest) {
-        @SuppressWarnings("squid:S2245")
-        Random random = new Random(onboardingRequest.getUserId().hashCode());
-        return random;
+    private Mono<OnboardingDTO> invokePdndServices(String accessToken, OnboardingDTO onboardingRequest, String fiscalCode, PdndServicesInvocation pdndServicesInvocation, Message<String> message) {
+        Mono<Optional<Object>> inpsInvocation = pdndServicesInvocation.requireInpsInvocation()
+                ? Mono.just(Optional.of("TODO Call INPS Service"))
+                : Mono.just(Optional.empty());
+
+        Mono<Optional<RispostaE002OKDTO>> anprInvocation = pdndServicesInvocation.requireAnprInvocation()
```

Comment:
Create a @Service specialized into invoke and extract Anpr data and move there this logic and the logic on lines 128-147 (on a separate method, that will be called instead of that lines)